src/basic/time-util.c

```diff
@@ -1020,14 +1023,15 @@ int parse_time(const char *t, usec_t *usec, usec_t default_unit) {
                 if (*e == '.') {
                         char *b = e + 1;
 
+                        if (*b == '-') /* Don't allow 0.-0 */
+                                return -EINVAL;
+
                         errno = 0;
                         z = strtoll(b, &e, 10);
                         if (errno > 0)
                                 return -errno;
-
                         if (z < 0)
```

Comment:
This is dead code now, since you're checking for a starting '-' earlier... Perhaps assert(z >= 0) would be better? Same with the if (l < 0) check a few lines above.